include/pybind11/numpy.h

```diff
@@ -1482,6 +1482,54 @@ struct vectorize_arg {
     using type = conditional_t<vectorize, array_t<remove_cv_t<call_type>, array::forcecast>, T>;
 };
 
+
+template <typename Func, typename Return, typename... Args>
```

Comment:
Maybe add a comment above each version of `vectorize_returned_array`, to explain what it does? It's not that obvious from quick glance without the context of the PR.